File: src/main/java/com/javahere/reachyourgoal/service/UserService.java
package com.javahere.reachyourgoal.service;

import com.javahere.reachyourgoal.dto.UserDTO;
import com.javahere.reachyourgoal.dto.UserDTOLogin;
import org.springframework.security.core.userdetails.UserDetailsService;

public interface UserService extends UserDetailsService {

    Boolean usernameExists(String username);

    Boolean emailExists(String email);

    UserDTO createUser(UserDTO userDTO);

    Boolean login(UserDTOLogin userDTOLogin);
}

File: src/main/kotlin/me/javahere/reachyourgoal/service/UserService.kt
package me.javahere.reachyourgoal.service

import me.javahere.reachyourgoal.domain.dto.UserDto
import me.javahere.reachyourgoal.domain.dto.request.RequestRegister
import me.javahere.reachyourgoal.domain.dto.request.RequestUpdateEmail
import me.javahere.reachyourgoal.domain.id.UserId

interface UserService {
    suspend fun registerProductionMode(user: RequestRegister)

    suspend fun registerDevelopMode(user: RequestRegister): String

    suspend fun confirmRegister(token: String): UserDto

    suspend fun confirmNewEmail(token: String): UserDto

    suspend fun findUserById(userId: UserId): UserDto

    suspend fun findUserByEmail(email: String): UserDto

    suspend fun findUserByUsername(username: String): UserDto

    suspend fun updateUser(
        userId: UserId,
        firstName: String? = null,
        lastName: String? = null,
    ): UserDto

    suspend fun updateEmail(request: RequestUpdateEmail)

    suspend fun deleteUserById(userId: UserId)
}
